packages/wp-source/src/libraries/handlers/__tests__/post-type.tests.ts

```diff
@@ -22,9 +22,14 @@ beforeEach(() => {
 describe("postType", () => {
   test("returns 404 if not found", async () => {
     // Mock Api responses
-    api.get = jest.fn().mockResolvedValue(mockResponse([]));
+    api.get = jest.fn(_ => Promise.resolve(mockResponse([])));
+
+    // TODO: The test fails if we use this form instead 🤷‍♂️:
+    // api.get = jest.fn().mockResolvedValue(mockResponse([]));
```

Comment:
The difference is that in the first case you instantiate a new response everytime `api.get` is executed. In the second case it returns the same instance, and it fails because `response.json()` can only be run once (which is weird).

When I implemented this part, my idea was to use the response object to be able to access its headers in addition to the body object, but it turns out not to be a good decission, I guess.  😅

Reply:
right, thanks for the explanation 🙂 I ll update the comment then.